### PR TITLE
Add support of Mardown Exporter

### LINKED ISSUE #2532: New exporter: markdown table
**Is your feature request related to a problem or area of OpenRefine? Please describe.**

It would be useful to have an exporter to represent a table in Markdown. The format is specified in the Github-Flavoured Markdown guide:
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables

And in the specs: https://github.github.com/gfm/#tables-extension-

**Describe the solution you'd like**

Given

| Field   | Data                  |
|---------|-----------------------|
| Name    | Galanthus nivalis     |
| Color   | White                 |
| IUCN ID | 162168                |
| Name    | Narcissus cyclamineus |
| Color   | Yellow                |
| IUCN ID | 161899                |

we would get

```markdown
| Field   | Data                  |
|---------|-----------------------|
| Name    | Galanthus nivalis     |
| Color   | White                 |
| IUCN ID | 162168                |
| Name    | Narcissus cyclamineus |
| Color   | Yellow                |
| IUCN ID | 161899                |
```

Note the need to add spaces to pad the cell values to the same length.

**Describe alternatives you've considered**
I don't think the custom tabular exporter has support for this sort of cell padding. If we add support for padding in it, we could potentially reuse it. But the export workflow would be more cumbersome (since the exporter would need to be configured every time).

**Additional context**

This could be based on an existing Java library, such as https://github.com/Steppschuh/Java-Markdown-Generator

If we are to include examples of tables in our Markdown documentation, this could be very useful.

### PR DESCRIPTION
**Pull request by Unnamed-1408**
Fixes #2532

Changes proposed in this pull request:
- Add Markdown options to `Export` menu
- Add Markdown exporter
- Add Markdown exporter test file

### CODE REVIEW

File: main/src/com/google/refine/exporters/ExporterRegistry.java [hunk 1]
```diff
@@ -23,8 +23,8 @@
 A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
```
Comment (antoine2711) [inline main/src/com/google/refine/exporters/ExporterRegistry.java:26]: There is only spaces change in that file. It should be restored, no need to make noise on our history.

**Comment by Unnamed-1408** [comment]
I'm sorry about the spaces change, it seems like the IDEA automatically modified these files. Should I remove the spaces change?

**Review by wetneb — APPROVED** [review]
Sorry about the delay in reviewing this. It looks good to me, thank you!

**Comment by Unnamed-1408** [comment]
Thanks for merging this pr, but I found that the cypress test I wrote had an error (ui_test chrome) during testing in `main/tests/cypress/cypress/integration/project/project-header/export_project.spec.js`, but I have no idea how to fix, sorry for that

**Comment by Unnamed-1408** [comment]
Thanks for merging this pr, but I found that the cypress test I wrote had an error (ui_test chrome) during testing in `main/tests/cypress/cypress/integration/project/project-header/export_project.spec.js`, but I have no idea how to fix, sorry for that

**Comment by wetneb** [comment]
Ah, thanks for looking! Feel free to submit another PR to fix that :)

**Comment by wetneb** [comment]
I have temporarily reverted this PR to keep the CI green - feel free to revert my revert in your fixup PR :)

**Pull request merged**